common/src/main/java/org/opensearch/ml/common/agent/MLAgent.java

```diff
@@ -246,6 +285,9 @@ public static MLAgent parse(XContentParser parser) throws IOException {
                 case APP_TYPE_FIELD:
                     appType = parser.text();
                     break;
+                case IS_HIDDEN_FIELD:
+                    isHidden = parser.booleanValue();
+                    break;
```

Comment:
Would this allow users to pass isHidden field through API?

Reply:
Even if they pass the field, that will be replaced with our code logic:

```
if (llmId != null) {
                String[] excludes = new String[] { MLModel.MODEL_CONTENT_FIELD, MLModel.OLD_MODEL_CONTENT_FIELD };
                mlModelManager.getModel(llmId, null, excludes, ActionListener.wrap(mlModel -> {
                    processModelAndRegisterAgent(agent, mlModel.getIsHidden(), now, listener, context);
                }, e -> {
                    log.error("Failed to retrieve the ML model with ID: " + llmId, e);
                    listener.onFailure(e);
                }));
            } else {
                processModelAndRegisterAgent(agent, false, now, listener, context);
            }
```

Reply:
Suggest not parse this field to avoid some confusion. Someone in future may think we can allow user set this field and mess up the code

Reply:
It is better to not allow isHidden parameter through API and set it based on super admin check.

Reply:
done